### ISSUE TITLE
`osrm-datastore` should release locks on crashes, if possible

### ISSUE THREAD

**Issue by danpat** [Project-OSRM/osrm-backend#3477]
We encountered a situation where `osrm-datastore` crashed due to an out-of-memory condition.

Because `osrm-datastore` uses interprocess locks (`boost::interprocess::named_sharable_mutex`), these persist after the process dies.  If we crash without releasing the lock, it hangs around, causing later invocations believing a process is still running, leading to an incorrect report of `A data update is in progress`.

The `osrm-datastore` tool should try really hard to clean up on exit.  We should probably go as far as:

  - http://www.cplusplus.com/reference/exception/set_terminate/
  - adding signal handlers

Basically, we should always unlock, for everything except `SIGKILL`.

**Comment by TheMarex** [comment]
This was fixed 🎉
